### PR TITLE
Dropdown: anchor popover to the dropdown wrapper (instead of the toggle)

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Part of #42770

This PR:
 - fixes the issue flagged in https://github.com/WordPress/gutenberg/pull/43327
 - follows up the work done in https://github.com/WordPress/gutenberg/pull/42989
 - partially reverts some changes done in https://github.com/WordPress/gutenberg/pull/41361 (in particular how the `Dropdown` component picks its fallback popover anchor)

## Why?

See https://github.com/WordPress/gutenberg/pull/43327#issuecomment-1219248496

## How?

The main fix is that the `Dropdown` component switches **back to using its wrapper/container as the `Popover` anchor** (instead of the toggle).

The rest of the changes are needed to make `CustomGradientPicker` work with the new dropdown anchor — in particular, some style changes were necessary to make sure that **the dropdown wrapper for control and insert points were positioned and sized correctly**, since they are not used as the popover anchors.

## Testing Instructions

- Make sure that the issue flagged in https://github.com/WordPress/gutenberg/pull/43327 is fixed on this PR
- Make sure that the `CustomGradientPicker` component still works as expected (in particular, clicking on the gradient bar on existing control points, or inserting a new control point should reveal a color picker popover positioned like in `trunk`)
- In Storybook, make sure that the issue flagged in https://github.com/WordPress/gutenberg/pull/43327#issuecomment-1219253506 can't be reproduced on this PR (you will need to apply the changes from this PR to the Storybook example)
- In Storybook, check all components using `Dropdown` and `DropdownMenu` and make sure there aren't regressions compared to [`trunk`](https://wordpress.github.io/gutenberg/)
- In the editor, make sure all dropdowns keep working as expected

### `DropdownMenu` in Storybook

`trunk`:

https://user-images.githubusercontent.com/1083581/185390346-45436e0d-ce60-4c16-b87e-e3dee2074591.mp4

This PR:

https://user-images.githubusercontent.com/1083581/185389871-e047fe55-c971-48b1-bf0e-54f0df40360a.mp4

### "More" menu in the editor

`trunk`:

https://user-images.githubusercontent.com/1083581/185391212-26439966-2baa-4e91-b301-9d039a7ffb71.mp4

This PR:

https://user-images.githubusercontent.com/1083581/185391149-41887a8c-b3ce-43ee-a94a-e0264e33e597.mp4

### CODE REVIEW

File: packages/components/src/dropdown/index.js
Comment (ciampo) [inline packages/components/src/dropdown/index.js:115]: This is the main change that introduces the fix

File: packages/components/src/custom-gradient-picker/style.scss
Comment (ciampo) [inline packages/components/src/custom-gradient-picker/style.scss:69]: This was not necessary anymore, since `components-custom-gradient-picker__insert-point-dropdown` also sets the width

File: packages/components/src/custom-gradient-picker/style.scss
Comment (ntsekouras) [inline packages/components/src/custom-gradient-picker/style.scss:26]:
```suggestion
		// Trim any internal white spacing that would be caused by inline positioned elements.
```

**Review by ntsekouras — APPROVED** [review]
Great job Marco! Tested and looks good, thanks!

**Comment by ciampo** [comment]
Since this change affects a lot of components in Gutenberg (every dropdown!), it'd be great if we could also get a second pair of eyes checking for any potential regressions.

Thank you!

**Review by andrewserong — APPROVED** [review]
Nice work @ciampo, and good idea reverting part of the change from #41361, I think in general that makes the change here feel reasonably safe — and moving the styling to the dropdown wrapper instead of the button sounds good to me 👍

The only direct issue I ran into was with the Storybook example of the gradient picker, where the Popover appeared behind some of the input fields. I couldn't replicate this in the editor, though, so only encountered it in Storybook. (In the editor it's difficult to get the Popover into a position where it covers the controls, but when I did manage to do that, the Popover was correctly on top of the input fields)

image

To try to test as exhaustively as I could, I tested the below components in the post and site editors and they were working well for me: (I found these by searching the code base for `<Dropdown>` but might have missed a couple of obscure components here or there)

<details>

<summary>Tested components</summary>

✅ `block-alignment-matrix-control`
✅ `block-navigation`
✅ `block-settings-menu`
✅ `block-switcher`
✅ `block-variation-transforms`
✅ `color-style-selector`
✅ `colors-gradients`
✅ `duotone-control`
✅ `inserter`
✅ `media-replace-flow`
✅ `preview-options`
✅ `publish-date-time-picker`
✅ `format-toolbar`
✅ `tool-selector`
✅ `post-date`
✅ `query-toolbar`
✅ `video` block tracks dropdown
✅ `border-control-dropdown`
✅ `circular-option-picker`
✅ `color-palette`
✅ `item-group`
✅ `navigator`
✅ `palette-edit`
✅ `toolbar`
✅ `toolbar-dropdown-menu`
✅ `toolspanel`
✅ `block-placeholder`
✅ `post-schedule`
✅ `post-template`
✅ `post-url`
✅ `post-visibility`
✅ `add-new-template`
✅ `document-actions`
✅ `actions`
✅ `sidebar`
✅ `template-card`
✅ `template-details`
✅ `table-of-contents`
✅ `more-menu-dropdown`

</details>

Overall, all looking good, with a small caveat:

I ran into an issue with testing the Image block buttons, but fortunately that's been fixed in https://github.com/WordPress/gutenberg/pull/43361 — looks like this PR will need a rebase just to make sure those still work. Once it's rebased, we'd then just want to double check that the aspect ratio dropdown in the toolbar is positioned correctly, then I think we'd be good to go.

Otherwise, this PR looks good to go to me, I think we can always look into the Popover z-index in the gradient picker storybook as a separate task, if you wanted to address that separately to unblock merging.

Happy to test further on Monday if need be!

**Comment by ciampo** [comment]
Thank you @andrewserong for your thorough testing!

> The only direct issue I ran into was with the Storybook example of the gradient picker, where the Popover appeared behind some of the input fields. I couldn't replicate this in the editor, though, so only encountered it in Storybook. (In the editor it's difficult to get the Popover into a position where it covers the controls, but when I did manage to do that, the Popover was correctly on top of the input fields)

That is a known issue, which I reported first in https://github.com/WordPress/gutenberg/issues/41390 and recently in https://github.com/WordPress/gutenberg/issues/42994 (should be unrelated to the changes in this PR)

> I ran into an issue with testing the Image block buttons, but fortunately that's been fixed in https://github.com/WordPress/gutenberg/pull/43361 — looks like this PR will need a rebase just to make sure those still work. Once it's rebased, we'd then just want to double check that the aspect ratio dropdown in the toolbar is positioned correctly, then I think we'd be good to go.

Good idea — i've just rebased

**Comment by ciampo** [comment]
The aspect ratio dropdown looks good after the rebase

Screenshot 2022-08-19 at 09 46 17

I'm going to wait for CI to pass ✅  and likely go ahead and merge

**Pull request merged**
